pkg/clientcert/cert_controller.go

```diff
@@ -109,8 +108,49 @@ type clientCertificateController struct {
 func NewClientCertificateController(
 	clientCertOption ClientCertOption,
 	csrOption CSROption,
-	hubCSRInformer certificatesinformers.CertificateSigningRequestInformer,
-	hubCSRClient csrclient.CertificateSigningRequestInterface,
+	hubCSRInformer certificatesinformers.Interface,
+	spokeSecretInformer corev1informers.SecretInformer,
+	spokeKubeClient kubernetes.Interface,
+	hubKubeClient kubernetes.Interface,
+	recorder events.Recorder,
+	controllerName string,
+) (factory.Controller, error) {
+	var csrCtrl csrControl = nil
+	if features.DefaultSpokeMutableFeatureGate.Enabled(features.V1beta1CSRAPICompatibility) {
+		v1CSRSupported, v1beta1CSRSupported, err := helpers.IsCSRSupported(hubKubeClient)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed CSR api discovery")
+		}
+		if !v1CSRSupported && v1beta1CSRSupported {
+			csrCtrl = &v1beta1CSRControl{
+				hubCSRInformer: hubCSRInformer.V1beta1().CertificateSigningRequests(),
+				hubCSRLister:   hubCSRInformer.V1beta1().CertificateSigningRequests().Lister(),
+				hubCSRClient:   hubKubeClient.CertificatesV1beta1().CertificateSigningRequests(),
+			}
+		}
+		klog.Info("Using v1beta1 CSR api to manage spoke client certificate")
```

Comment:
This log should be in the `if !v1CSRSupported && v1beta1CSRSupported {`  block

Reply:
nice catch